src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -3996,6 +3997,96 @@ test_mongoc_client_recv_network_error (void)
    mongoc_server_stream_cleanup (stream);
 }
 
+void
+test_mongoc_client_get_handshake_hello_response_single (void)
+{
+   mongoc_client_t *client;
+   mongoc_server_description_t *monitor_sd;
+   mongoc_server_description_t *invalidated_sd;
+   mongoc_server_description_t *handshake_sd;
+   bson_error_t error = {0};
+
+   client = test_framework_new_default_client ();
+   /* Perform server selection to establish a connection. */
+   monitor_sd = mongoc_client_select_server (
+      client, false /* for writes */, NULL /* read prefs */, &error);
+   ASSERT_OR_PRINT (monitor_sd, error);
+   BSON_ASSERT (
+      0 != strcmp ("Unknown", mongoc_server_description_type (monitor_sd)));
+
+   /* Invalidate the server. */
+   mongoc_topology_invalidate_server (client->topology, monitor_sd->id, &error);
+
+   /* Get the new invalidated server description from monitoring. */
+   invalidated_sd =
+      mongoc_client_get_server_description (client, monitor_sd->id);
+   BSON_ASSERT (NULL != invalidated_sd);
+   ASSERT_CMPSTR ("Unknown", mongoc_server_description_type (invalidated_sd));
+
+   /* The previously established connection should have a valid server
+    * description. */
+   handshake_sd = mongoc_client_get_handshake_description (
+      client, monitor_sd->id, NULL /* opts */, &error);
+   ASSERT_OR_PRINT (handshake_sd, error);
+   BSON_ASSERT (
+      0 != strcmp ("Unknown", mongoc_server_description_type (handshake_sd)));
+
+   mongoc_server_description_destroy (handshake_sd);
+   mongoc_server_description_destroy (invalidated_sd);
+   mongoc_server_description_destroy (monitor_sd);
+   mongoc_client_destroy (client);
+}
+
+void
+test_mongoc_client_get_handshake_hello_response_pooled (void)
+{
+   mongoc_client_t *client;
+   mongoc_server_description_t *monitor_sd;
+   mongoc_server_description_t *invalidated_sd;
+   mongoc_server_description_t *handshake_sd;
+   bson_error_t error = {0};
+   bool ret;
+
+   client = test_framework_new_default_client ();
```

Comment:
What about this test is pooled? You're fetching a `client` in the same manner that was done in `test_mongoc_client_get_handshake_hello_response_single` above.

Reply:
🤦 I totally missed that. Updated to use a client pool.